src/gleam/regex.gleam

```diff
@@ -222,12 +218,40 @@ pub fn scan(with regex: Regex, content string: String) -> List(Match) {
   do_scan(regex, string)
 }
 
-if erlang {
-  external fn do_scan(Regex, String) -> List(Match) =
-    "gleam_stdlib" "regex_scan"
-}
+@target(erlang)
+external fn do_scan(Regex, String) -> List(Match) =
+  "gleam_stdlib" "regex_scan"
+
+@target(javascript)
+external fn do_scan(Regex, String) -> List(Match) =
+  "../gleam_stdlib.mjs" "regex_scan"
 
-if javascript {
-  external fn do_scan(Regex, String) -> List(Match) =
-    "../gleam_stdlib.mjs" "regex_scan"
+// Creates a new `String` by replacing all occurrences of a given Regex.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > let assert Ok(re) = regex.from_string("[.]+")
+/// > regex.replace("www.example.com", each: re, with: "-", global: True)
+/// "www-example-com"
+/// > regex.replace("www.example.com", each: re, with: "-", global: False)
+/// "www-example.com"
+/// ```
+///
+pub fn replace(
+  in string: String,
+  each pattern: Regex,
+  with substitute: String,
+  global replace_all: Bool,
```

Comment:
Let's remove the global argument and just have it replace all for now.

Reply:
This would be a good application of optionals with defaults I guess? https://github.com/gleam-lang/gleam/discussions/1122

Reply:
it would